auth_server/authn/custom_authn.go

```diff
@@ -0,0 +1,82 @@
+/*
+   Copyright 2019 Cesanta Software Ltd.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       https://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package authn
+
+import (
+	"fmt"
+	"github.com/cesanta/glog"
+	"plugin"
+)
+
+type CustomAuthnConfig struct {
+	Configured  string `yaml:"configured"`
+	Plugin_path string `yaml:"plugin_path"`
+}
+
+func (c *CustomAuthnConfig) Validate() error {
+	if c.Configured != "true" {
+		return fmt.Errorf("custom_auth should set to true")
+	}
+	if c.Plugin_path == "" {
```

Comment:
check that the file exists as well. in fact, perform the symbol type check as well.

Reply:
I added the file check. I did not get the point to perform the symbol type check. Is this about file extension or x.(T) check?